scripts/train/train.py

```diff
@@ -149,6 +150,97 @@ def build_composer_model(model_cfg: DictConfig,
     return COMPOSER_MODEL_REGISTRY[model_cfg.name](model_cfg, tokenizer)
 
 
+# TODO cleanup / decompose logic
+def log_analytics_details(mosaicml_logger: MosaicMLLogger,
+                          model_config: DictConfig,
+                          train_loader_config: DictConfig,
+                          eval_loader_config: Union[DictConfig, ListConfig,
+                                                    None],
+                          callback_configs: Union[DictConfig,
+                                                  None], tokenizer_name: str,
+                          load_path: Union[str, None], save_folder: Union[str,
+                                                                          None],
+                          icl_tasks_config: Optional[Union[ListConfig, str]],
+                          eval_gauntlet: Optional[Union[DictConfig, str]]):
+    metrics = {
+        'llmfoundry/tokenizer_name':
+            tokenizer_name,
+        'llmfoundry/llmfoundry_run_type':
+            'training',
+        'llmfoundry/train_loader_name':
+            train_loader_config.get('name'),
+        'llmfoundry/train_loader_workers':
+            train_loader_config.get('dataset').get('num_workers'),
+    }
+
+    if callback_configs is not None:
+        metrics['llmfoundry/callbacks'] = [
+            name for name, _ in callback_configs.items()
+        ]
+
+    if eval_gauntlet is not None:
+        metrics['llmfoundry/gauntlet_configured'] = True
+
+    if icl_tasks_config is not None:
+        if isinstance(icl_tasks_config, str):
+            metrics['llmfoundry/icl_configured'] = True
+        elif len(icl_tasks_config) > 0:
+            metrics['llmfoundry/icl_configured'] = True
+
+    # TODO: what's a good name for this? ideally we want the keys to be the same for
+    # training and eval to make querying easier.
+    if load_path is not None:
+        metrics['llmfoundry/cloud_provider_loading'] = load_path.split(':')[0]
+    if save_folder is not None:
+        metrics['llmfoundry/cloud_provider_saving'] = save_folder.split(':')[0]
+
+    if train_loader_config.get('dataset').get('hf_name', None) is not None:
+        metrics['llmfoundry/train_dataset_hf_name'] = train_loader_config.get(
+            'dataset').get('hf_name')
+    if train_loader_config.get('name') == 'finetuning':
+        metrics['llmfoundry/llmfoundry_run_subtype'] = 'IFT (finetuning)'
```

Comment:
What if we make these consistent with "task types" (from finetuning)
```suggestion
        metrics['llmfoundry/train_task_type'] = 'INSTRUCTION_FINETUNE'
```
& CONTINUED_PRETRAIN + PRETRAIN